### ISSUE TITLE
Different probability in demo website and run_placesCNN_unified.py for SCENE CATEGORIES

### ISSUE THREAD

**Issue by rahulranjan07** [CSAILVision/places365#45]
Does anyone know why the demo here: http://places2.csail.mit.edu/demo.html and the run_placesCNN_unified.py . I tried testing with the same images mentiones in the README.md file, and the result differs a lot:

According to the README file and places demo website, the result is as follows:

> RESULT ON http://places.csail.mit.edu/demo/6.jpg
    --TYPE: indoor
    --SCENE CATEGORIES:
    0.690 -> food_court
    0.163 -> cafeteria
    0.033 -> dining_hall
    0.022 -> fastfood_restaurant
    0.016 -> restaurant
    --SCENE ATTRIBUTES:
    no horizon, enclosed area, man-made, socializing, indoor lighting, cloth, congregating, eating, working
    Class activation map is output as cam.jpg

But when I run run_placesCNN_unified.py on my server, I get the following result:

> RESULT ON http://places.csail.mit.edu/demo/6.jpg
--TYPE OF ENVIRONMENT: indoor
--SCENE CATEGORIES:
0.511 -> food_court
0.085 -> fastfood_restaurant
0.083 -> cafeteria
0.040 -> dining_hall
0.021 -> flea_market/indoor
--SCENE ATTRIBUTES:
no horizon, enclosed area, man-made, socializing, indoor lighting, cloth, congregating, eating, working
Class activation map is saved as cam.jpg

Does anyone knows which model the current demo website is using ?

**Comment by nickspriet** [comment]
I'm having the same issue. Running the `run_placesCNN_unified.py` code gives me the following prediction:
0.232 -> beauty_salon
**0.204 -> reception**

while running the `demo` on the `Places365 website` gives the following:
**0.613 -> reception**

Why is this difference so big? I'm using the same `wideresnet18_places365.pth.tar` model as shown in the README file.

**Comment by metalbubble** [comment]
The weight is converted once due to the upgrade of the pytorch0.4, so there might be some small numeric change in the weight value. I don't think it affects the overall performance of the model.

**Comment by nickspriet** [comment]
And which network architecture is used in the [online demo model](http://places2.csail.mit.edu/demo.html)?

**Comment by MasterYodah** [comment]
Hi @metalbubble , I ran into same issue today. After a closer inspection I found the problem to be [this line](https://github.com/CSAILVision/places365/blob/0725d607eabe2f37fdc6c7481fab14ffef0079d3/run_placesCNN_unified.py#L132). 

Could you explain why you enforce transition weights to be non-negative? Intuitively this prevents any pooled features from directly diminishing output probabilities (discounting softmax). but is there any specific reason behind this?
